src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenuLink.razor.cs

```diff
@@ -91,22 +94,21 @@ public partial class FluentNavMenuLink : FluentComponentBase
 
     internal bool HasIcon => !string.IsNullOrWhiteSpace(Icon) || IconContent is not null;
 
-    [CascadingParameter(Name = "NavMenuExpanded")]
-    private bool NavMenuExpanded { get; set; }
-
     public FluentNavMenuLink()
     {
         Id = Identifier.NewId();
     }
 
-    protected override void OnParametersSet()
+    protected override void OnInitialized()
     {
+        base.OnInitialized();
         NavMenu.AddNavMenuLink(this);
+    }
 
+    protected override void OnParametersSet()
```

Comment:
Override SetParametersAsync instead of OnParametersSet, so we can track before/after values without having to have additional private fields to store the previous state.